server/pulp/server/webservices/views/content.py

```diff
@@ -275,6 +275,28 @@ def _add_repo_memberships(units, type_id):
             unit['repository_memberships'] = list(association_map.get(unit['_id'], []))
         return units
 
+    def _parse_args(cls, args):
+        """
+        Ensures that the "include_repos" argument gets converted to a bool if necessary. Parent
+        class doc block follows:
+
+        Some search views are required to be able to include extra information. This function
+        separates those flags from the search arguments.
+
+        :param args: Search parameters mixed with extra options. Any extra options should be
+                     removed and returned separately.
+        :type  args: django.http.QueryDict
+
+        :return: filtered search parameters and options
+        :rtype:  tuple containing a 2 dicts
+        """
+        search_params, options = super(ContentUnitSearch, cls)._parse_args(args)
+
+        # if this was a GET request, the argument will be a string. We convert it to a bool here.
+        if isinstance(options.get('include_repos'), basestring):
```

Comment:
I think we could this logic into the base search class and not need to override _parse_args. The bool issue will be present for all of the search endpoints that have the optional params separated in _parse_args.